src/components/Initiatives/InitiativeDetailsPage/InitativeFilterDialog/InitiativeFilterDialog.tsx

```diff
@@ -0,0 +1,182 @@
+        actionItem.componentRef === componentRef &&
+        actionItem.rule.expression === rule,
+    );
+
+    return passed === pass;
+  };
+};
+
+interface InitiativeFilterDialogProps {
+  handleClose: () => void;
+  initiative: Initiative;
+  actionItems: InitiativeActionItem[];
+  isOpen: boolean;
+  setFilter: (filter: InitiativeFilter) => void;
+}
+
+const InitiativeFilterDialog = ({
+  initiative,
+  isOpen,
+  actionItems,
+  handleClose,
+  setFilter,
+}: InitiativeFilterDialogProps) => {
+  const classes = useStyles();
+  const { checkedFilters, oneOf } = useFilter();
+
+  const ruleFilterDefinitions = useMemo(() => {
+    return mapValues(
+      mapByString(initiative.rules, rule => `${rule.ruleId}`),
+      rule => {
+        return {
+          display: ruleName(rule),
+          value: rule.expression,
+          id: rule.ruleId.toString(),
+        };
+      },
+    );
+  }, [initiative.rules]);
+
+  const { filterGroups, loading } = useFilters(
+    (entityRef: string) => entityRef,
+    {
+      baseFilters: groupAndSystemFilters,
+    },
+  );
+
+  const filtersDefinition = [
+    {
+      name: 'Failing rules',
+      filters: ruleFilterDefinitions,
+      generatePredicate: (failingRule: string) =>
+        createRulePredicate(failingRule, actionItems, false),
+    },
+    {
+      name: 'Passing rules',
+      filters: ruleFilterDefinitions,
+      generatePredicate: (passingRule: string) =>
+        createRulePredicate(passingRule, actionItems, true),
+    },
+    ...(filterGroups ?? []),
+  ];
+
+  const handleSaveFilters = () => {
+    const allFilters: InitiativeFilter[] = [];
+
+    filtersDefinition.forEach((filterDefinition, idx) => {
+      const predicateFilters = toPredicateFilters(
+        checkedFilters,
+        filterDefinition.name,
+      );
+      const filterOneOf = oneOf[filterDefinition.name] ?? true;
+
+      allFilters[idx] = (componentRef: string) => {
+        const results = Object.keys(predicateFilters)
+          .filter(id => predicateFilters[id])
+          .map(id =>
+            filterDefinition.generatePredicate(
+              filterDefinition.filters[id].value,
+            )(componentRef),
+          );
+
+        if (results.length === 0) {
+          return true;
+        }
+
+        return filterOneOf ? results.some(Boolean) : results.every(Boolean);
+      };
+    });
+
+    setFilter(combinePredicates(Object.values(allFilters)));
+
+    handleClose();
+  };
+
+  return (
+    <Dialog open={isOpen} onClose={handleClose}>
+      <DialogContent className={classes.dialogContent}>
+        {loading ? (
+          <Progress />
+        ) : (
+          <FilterCard
+            filterDefinitions={filtersDefinition}
+            title="Filter Initiative"
+          />
+        )}
+      </DialogContent>
+      <DialogActions>
+        <Button
+          onClick={handleSaveFilters}
+          color="primary"
+          aria-label="Close modal"
+        >
+          Close
```

Comment:
Seems weird that this is labeled "close" but seems to really be "save" or "apply filters" -- could we update the copy?